p2p/connectionpool/connectionpool.go

```diff
@@ -148,21 +152,23 @@ func (cp *ConnectionPool) handleNewConnection(rPub p2pcrypto.PublicKey, newConn
 		} else { // newConn < curConn
 			cp.net.Logger().Warning("connection created while connection already exists between peers, closing new connection. existing session ID=%v, new session ID=%v, remote=%s", curConn.Session().ID(), newConn.Session().ID(), rPub)
 			closeConn = newConn
+			err = errors.New("closing connection")
```

Comment:
why do you need this error? also, it is not as elaborate as the warning above, if the returned error is printed as well I'd recommend having the content of the warning inside the error message itself

Reply:
If we're handling a new connection, be it from out or inbound, if we already have a connection with the same peer stored we want to indicate that adding the connection was failed if we kept the old connection. this indicates higher levels that no further action needs to be done with this connection.

Reply:
I will return the warning content as error